File: Single_Number_II/singleNumber.cpp
class Solution {
public:
    int singleNumber(int A[], int n) {
        int result = 0;
        int i = 0;
        int j = 0;
        int count = 0;
        //sum every bit for A array
        for (i = 0; i < 32; i++){
            for (j = 0; j < n; j++){
                if ((A[j] >> i) & 1){
                    count++;
                }
            }
            
            if (count % 3 == 1){
                result |= 1 << i;
            }
            count = 0;
        }
        
        return result;
    }
};

File: Single_Number_II/singleNumber.py
class Solution:
    # @param A, a list of integer
    # @return an integer
    def singleNumber(self, A):
        B = {}
        for num in A:
            if B.has_key(num):
                B[num] += 1
            else:
                B[num] = 1
                
        for num in B.keys():
            if B[num] == 1:
                return num